marketplaceItems/audioFocus/appResources/appData/audioFocus_app.js

```diff
@@ -0,0 +1,459 @@
+    }
+
+
+    // Disables all the pointers created and the mapping 
+    function disable() {
+        Controller.disableMapping(MAPPING_NAME);
+    }
+
+
+    // Returns the right UUID based on hand triggered
+    function getUUIDFromLaser(hand){
+        hand = hand === Controller.Standard.LeftHand 
+            ? Controller.Standard.LeftHand
+            : Controller.Standard.RightHand;
+        
+        var pose = getControllerWorldLocation(hand);
+        var start = pose.position;
+        var direction = Vec3.multiplyQbyV(pose.orientation, {x:0, y:1, z: 0});
+
+        var avatarIntersection = AvatarList.findRayIntersection({origin:start, direction:direction});
+        
+        var uuid = avatarIntersection.avatarID;
+
+        return uuid;
+    }
+
+
+    var MAPPING_NAME = "SOLO_POINTER";
+    var mapping = Controller.newMapping(MAPPING_NAME);
+
+    mapping.from(Controller.Hardware.Keyboard.LeftMouseButton).to(function (value) {
+        if (value === 1 || HMD.active) {
+            return;
+        }
+
+        var pickRay = Camera.computePickRay(lastMouseX, lastMouseY);
+        var avatarIntersection = AvatarList.findRayIntersection(pickRay);
+
+        var uuid = avatarIntersection.avatarID;
+
+        if (uuid && uuid !== MyAvatar.sessionUUID) {
+            soloAvatar(avatarIntersection.avatarID);
+        }
+    });
+
+
+    mapping.from(Controller.Standard.LTClick).to(function (value) {
+        if (value === 0) {
+            return;
+        }
+        
+        var uuid = getUUIDFromLaser(Controller.Standard.LeftHand);
+
+        if (uuid && uuid !== MyAvatar.sessionUUID) {
+            soloAvatar(uuid);
+        }
+    });
+
+
+    mapping.from(Controller.Standard.RTClick).to(function (value) {
+        if (value === 0) {
+            return;
+        }
+        
+        var uuid = getUUIDFromLaser(Controller.Standard.RightHand);
+
+        if (uuid && uuid !== MyAvatar.sessionUUID) {
+            soloAvatar(uuid);
+        }
+    });
+
+
+    // #endregion
+    // *************************************
+    // STOP MAPPING FUNCTIONS
+    // *************************************
+
+
+    // *************************************
+    // START SOLO FUNCTIONS
+    // *************************************
+    // #region Solo
+
+
+    // Adds avatar to the solo list
+    function addSolo(targetUUID) {
+        Audio.addToSoloList([targetUUID]);
+        updateUI();
+    }
+
+
+    // Remove Avatar from the solo list
+    function removeSolo(targetUUID) {
+        Audio.removeFromSoloList([targetUUID]);
+        updateUI();
+    }
+
+
+    // Remove all avatars from the solo list
+    function resetSolo() {
+        Audio.resetSoloList();
+        removeAllOverlays();
+        soloAvatars = {};
+        updateUI();
+    }
+
+    
+    // Add an avatar to the list and give them an overlay
+    function addAvatarToList(avatarUUID, displayUsername) {
+        soloAvatars[avatarUUID] = {
```

Comment:
It looks like you're duplicating the avatar's UUID here: it's both a `key` in the `soloAvatars` object, and it's also the value of the `id` `key` inside that first `key`'s value. please reduce complexity by only associating a `key` with a `name` (this'll require some minor changes elsewhere)